Create a guide for understanding the different ROS documentation sites (#1952)
ros2/ros2_documentation: source/Docs-Guide.rst

Docs Guide
==========

This guide explains the ROS 1 and ROS 2 documentation infrastructure.

> It should be helpful in understanding where specific resources live, how to ask questions, and which sites are being maintained.

Reviewer: Sorry to nitpick, but I just realized that this is somewhat of a change in tense (or, at least, in tone).  I think this sentence would be better as:

```suggestion
It is helpful in understanding where specific resources live, how to ask questions, and which sites are maintained.
```

(the same goes for the copy of this in index.rst)

Author: No worries, good catch. I updated this in 261287c.

   :local:

This guide explains the ROS 1 and ROS 2 documentation infrastructure.

> It should be helpful in understanding where specific resources live, how to ask questions, and which sites are being maintained.

Reviewer: I'll suggest we move this introduction before the Table of Contents.

Author: Good call - fixed in 5304928.

I also added a description of the Docs Guide in the "Where to start" section on the site's index page: cc35a1a.
